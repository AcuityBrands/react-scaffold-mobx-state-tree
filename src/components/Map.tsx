/**
 * Map
 * 
 * Creates contrived map component.  This component is currently
 * fairly limited in configurability and scope.
 */
import * as React from "react";
import mapboxgl = require('mapbox-gl');

interface IMapProps {
  height?: number
}

export default class Map extends React.Component<IMapProps, null> {

  private map:mapboxgl.Map;

  componentDidMount() {
    mapboxgl.accessToken = 'pk.eyJ1IjoiZGRicmFkc2hhdyIsImEiOiJQemtJRXhvIn0.zJBiAX860916-nQlwX839w';
    this.map = new mapboxgl.Map({
      container: 'map-container',
      style: 'mapbox://styles/mapbox/streets-v9',
      center: [-87.61620, 41.86600],
      zoom: 15.99,
      pitch: 40,
      bearing: 20
    });
    this.map.on('load', this.loadMapData);
  }

  recenterMap = () => {
    this.map.flyTo({
      center: [-87.61620, 41.86600],
      zoom: 15.99,
      pitch: 40,
      bearing: 0
    });
  }

  loadMapData = () => {
    this.map.addLayer({
      'id': 'room-extrusion',
      'type': 'fill-extrusion',
      'source': {
        'type': 'geojson',
        'data': '/data/3d-data.geojson'
      },
      'paint': {
        // See the Mapbox Style Specification for details on data expressions.
        // https://www.mapbox.com/mapbox-gl-js/style-spec/#expressions

        // Get the fill-extrusion-color from the source 'color' property.
        'fill-extrusion-color': ['get', 'color'],

        // Get fill-extrusion-height from the source 'height' property.
        'fill-extrusion-height': ['get', 'height'],

        // Get fill-extrusion-base from the source 'base_height' property.
        'fill-extrusion-base': ['get', 'base_height'],

        // Make extrusions slightly opaque for see through indoor walls.
        'fill-extrusion-opacity': 0.5
      }
    });
  }

  render() {
    return (
      <div id='map-container' style={{ height: this.props.height }}></div>
    );
  }
}