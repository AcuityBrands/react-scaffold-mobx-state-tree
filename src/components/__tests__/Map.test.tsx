import * as React from "react";
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import Map from '../Map';

Enzyme.configure({ adapter: new Adapter() });

const setup = propOverrides => {
  const props = Object.assign({}, propOverrides)
  const wrapper = Enzyme.shallow(<Map {...props} />)
  return {
    props,
    wrapper
  }
}

// This test suite uses a mock class that lives inside the __mocks__
// folder.  See https://facebook.github.io/jest/docs/en/manual-mocks.html
describe('Map Component', () => {
  // Minimal component test confirms component rendered
  it("can render", () => {
    const { wrapper } = setup({});
    expect(wrapper.exists()).toBe(true);
  })

  it("matches the snapshot", () => {
    const { wrapper } = setup({});
    expect(wrapper).toMatchSnapshot();
  })

  it("creates a root container div", () => {
    const { wrapper } = setup({});
    const div = wrapper.find("div");
    expect(div.length).toBeGreaterThan(0);
  })
})