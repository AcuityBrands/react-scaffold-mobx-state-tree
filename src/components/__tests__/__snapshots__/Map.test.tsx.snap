// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Map />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "id": "map-container",
      "style": Object {
        "height": undefined,
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "id": "map-container",
        "style": Object {
          "height": undefined,
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
